src/pages/Videoslisting/Videolisting.js

```diff
@@ -3,25 +3,31 @@ import {useData} from "../../contexts";
 
 function Videolisting(){
 
-    const {videos} = useData();    
+    const {filteredVideos,videosDispatch} = useData();    
 
     return(
         <div>
             <h1>Videos</h1>
-        <div className="videos-container">
-            {videos.map(item => 
-            <div className="video-card">
-                <div>
-                    <img src={item.thumbnail} className="thumbnail" alt={item.categoryName} />
-                    <span className="duration">{item.duration}</span>
-                </div>
-                <h3>{item.title} </h3>
-                <small>{item.views} views</small>
-                <div className="creator-details">
-                    <img src={item.avatar} className="avatar" alt="avatar" />
-                    <h3>{item.creator}</h3>
-                </div>       
+            <div className="filters">
+                <button className="filter-btn" onClick={() => videosDispatch({type:"FILTER_ALL"})}>All</button>
```

Comment:
Suggestion: you can create an array of categories and map them over. this will reduce the lines of code. and you can call `videoDispatch` once with respective categories